### PR TITLE
feat(shortmess): q fully hides recording message

### PR DESCRIPTION
**Pull request by trevarj**
When 'q' is set in the shortmess and cmdheight=1 it will now fully hide showing
if you are recording a macro instead of just shortening to "recording". This
  removes duplication when using `cmdheight=1` and `vim.fn.reg_recording()` in
  the statusline.

Relates to #19193

### CODE REVIEW

File: src/nvim/drawscreen.c [hunk 1]
```diff
@@ -1149,13 +1149,12 @@ void clearmode(void)
...
 {
-  msg_puts_attr(_("recording"), attr);
   if (shortmess(SHM_RECORDING)) {
     return;
   }
 
-  char s[4];
-  snprintf(s, ARRAY_SIZE(s), " @%c", reg_recording);
+  char s[13];
+  snprintf(s, ARRAY_SIZE(s), "recording @%c", reg_recording);
```
Comment (zeertzjq) [inline src/nvim/drawscreen.c:1157]: This message is not translated

Comment (trevarj) [reply src/nvim/drawscreen.c]: Thanks

File: runtime/doc/news.txt [hunk 2]
```diff
@@ -116,6 +116,8 @@ The following changes may require adaptations in user config or plugins.
   upstream tree-sitter and Helix to make it easier to share queries. The full
   list is documented in |treesitter-highlight-groups|.
 
+• |shm-q| completely hides recording macro message from last line when
```
Comment (zeertzjq) [inline runtime/doc/news.txt:119]: Also document this in vim_diff.txt

File: runtime/doc/news.txt [hunk 3]
```diff
@@ -116,6 +116,8 @@ The following changes may require adaptations in user config or plugins.
...
+  `cmdheight=1` instead of only shortening it.
```
Comment (zeertzjq) [inline runtime/doc/news.txt:120]: I don't think cmdheight=1 should be mentioned here. It's misleading.

File: test/old/testdir/test_registers.vim [hunk 4]
```diff
@@ -171,12 +171,12 @@ endfunc
 
 func Test_recording_status_in_ex_line()
   norm qx
-  redraw!
-  call assert_equal('recording @x', Screenline(&lines))
```
Comment (zeertzjq) [inline test/old/testdir/test_registers.vim:175]: Why is this moved below?

Comment (trevarj) [reply test/old/testdir/test_registers.vim:175]: Fixed.

File: test/old/testdir/test_registers.vim [hunk 5]
```diff
@@ -175,7 +175,7 @@ func Test_recording_status_in_ex_line()
   call assert_equal('recording @x', Screenline(&lines))
   set shortmess=q
   redraw!
-  call assert_equal('recording', Screenline(&lines))
+  call assert_equal('', Screenline(&lines))
```
Comment (zeertzjq) [inline test/old/testdir/test_registers.vim:178]:
```suggestion
  call assert_equal('', Screenline(&lines))  " Nvim: completely hides message
```

File: runtime/doc/options.txt [hunk 6]
```diff
@@ -5459,7 +5459,7 @@ A jump table for the options with a short description can be found at |Q_op|.
 		match", "Pattern not found", "Back at original", etc.
 	  C	don't give messages while scanning for ins-completion	*shm-C*
 		items, for instance "scanning tags"
-	  q	use "recording" instead of "recording @a"		*shm-q*
+	  q	do not show "recording @a" when recording a macro       *shm-q*
```
Comment (zeertzjq) [inline runtime/doc/options.txt:5462]:
```suggestion
	  q	do not show "recording @a" when recording a macro	
*shm-q*
```

File: runtime/lua/vim/_meta/options.lua [hunk 7]
```diff
@@ -5797,7 +5797,7 @@ vim.bo.sw = vim.bo.shiftwidth
 --- 	match", "Pattern not found", "Back at original", etc.
 ---   C	don't give messages while scanning for ins-completion	*shm-C*
 --- 	items, for instance "scanning tags"
----   q	use "recording" instead of "recording @a"		*shm-q*
+---   q	do not show "recording @a" when recording a macro       *shm-q*
```
Comment (zeertzjq) [inline runtime/lua/vim/_meta/options.lua:5800]:
```suggestion
---   q	do not show "recording @a" when recording a macro	
*shm-q*
```

File: src/nvim/options.lua [hunk 8]
```diff
@@ -7329,7 +7329,7 @@ return {
         	match", "Pattern not found", "Back at original", etc.
           C	don't give messages while scanning for ins-completion	*shm-C*
         	items, for instance "scanning tags"
-          q	use "recording" instead of "recording @a"		*shm-q*
+          q	do not show "recording @a" when recording a macro       *shm-q*
```
Comment (zeertzjq) [inline src/nvim/options.lua:7332]:
```suggestion
          q	do not show "recording @a" when recording a macro	
*shm-q*
```

File: runtime/doc/news.txt [hunk 9]
```diff
@@ -120,6 +120,7 @@ The following changes may require adaptations in user config or plugins.
   the default behavior of just refreshing the current buffer has been replaced by
   refreshing all buffers.
 
+• |shm-q| now completely hides recording macro message instead of only shortening it.
```
Comment (zeertzjq) [inline runtime/doc/news.txt:123]:
```suggestion
• |shm-q| now fully hides macro recording message instead of only shortening it.
```

File: runtime/doc/vim_diff.txt [hunk 10]
```diff
@@ -581,6 +581,9 @@ Autocommands:
 - |TermResponse| is fired for any OSC sequence received from the terminal,
   instead of the Primary Device Attributes response. |v:termresponse|
 
+Options:
+- |shm-q| completely hides recording macro message instead of only shortening it.
```
Comment (zeertzjq) [inline runtime/doc/vim_diff.txt:585]:
```suggestion
- |shm-q| fully hides macro recording message instead of only shortening it.
```

File: runtime/doc/options.txt [hunk 11]
```diff
@@ -5459,7 +5459,8 @@ A jump table for the options with a short description can be found at |Q_op|.
 		match", "Pattern not found", "Back at original", etc.
 	  C	don't give messages while scanning for ins-completion	*shm-C*
 		items, for instance "scanning tags"
-	  q	use "recording" instead of "recording @a"		*shm-q*
+	  q	do not show "recording @a" when recording a macro	
+*shm-q*
```
Comment (zeertzjq) [inline runtime/doc/options.txt:5463]:
```suggestion
	  q	do not show "recording @a" when recording a macro	*shm-q*
```

File: runtime/lua/vim/_meta/options.lua [hunk 12]
```diff
@@ -5797,7 +5797,8 @@ vim.bo.sw = vim.bo.shiftwidth
 --- 	match", "Pattern not found", "Back at original", etc.
 ---   C	don't give messages while scanning for ins-completion	*shm-C*
 --- 	items, for instance "scanning tags"
----   q	use "recording" instead of "recording @a"		*shm-q*
+---   q	do not show "recording @a" when recording a macro	
+*shm-q*
```
Comment (zeertzjq) [inline runtime/lua/vim/_meta/options.lua:5801]:
```suggestion
---   q	do not show "recording @a" when recording a macro	*shm-q*
```

File: src/nvim/options.lua [hunk 13]
```diff
@@ -7329,7 +7329,8 @@ return {
         	match", "Pattern not found", "Back at original", etc.
           C	don't give messages while scanning for ins-completion	*shm-C*
         	items, for instance "scanning tags"
-          q	use "recording" instead of "recording @a"		*shm-q*
+          q	do not show "recording @a" when recording a macro	
+*shm-q*
```
Comment (zeertzjq) [inline src/nvim/options.lua:7333]:
```suggestion
          q	do not show "recording @a" when recording a macro	*shm-q*
```
